### PR TITLE
Feature/google auth

### LINKED ISSUE #18: Auth logic

### PR DESCRIPTION
**Pull request by boqnd**

### CODE REVIEW

File: server/src/routes/authenticate.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+router.use(session({
+  resave: false,
+  saveUninitialized: true,
+  secret: 'SECRET'
+}));
+//
+
+//
+import passport from 'passport';
```
Comment (GritzMaze) [inline server/src/routes/authenticate.js]: Can we move the imports to the beginning?

File: server/src/routes/authenticate.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+var userProfile;
```
Comment (GritzMaze) [inline server/src/routes/authenticate.js]: Can we make this a `let`?

File: server/src/routes/authenticate.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  cb(null, user);
+});
+
+passport.deserializeUser(function(obj, cb) {
+  cb(null, obj);
+});
+//
+
+//
+import GoogleStrategy from 'passport-google-oauth';
```
Comment (GritzMaze) [inline server/src/routes/authenticate.js]: Can we move this to the top?

File: server/src/routes/authenticate.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+const GOOGLE_CLIENT_ID = '384846276379-ih4ucc8boll6b829up2vcrsiaolvirc2.apps.googleusercontent.com';
+const GOOGLE_CLIENT_SECRET = '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';
```
Comment (GritzMaze) [inline server/src/routes/authenticate.js:42]: Can we get this from the configs and make .env variables for them?

File: server/package.json [hunk 5]
```diff
@@ -33,13 +33,17 @@
     "convict": "^6.2.3",
     "cors": "^2.8.5",
     "dotenv": "^16.0.3",
+    "ejs": "^3.1.8",
```
Comment (GritzMaze) [inline server/package.json]: Do we need this?

**Review by GritzMaze — CHANGES_REQUESTED** [review]

File: server/src/routes/authenticate.js (hunk 4 above)
Comment (boqnd) [reply server/src/routes/authenticate.js:42]: Yes, we should. This is for test purposes only. Also we should discuss what google account will be used and with what permissions.

**Review by GritzMaze — APPROVED** [review]

**Review by Ilia-tod29 — APPROVED** [review]

**Pull request merged**